server/chess.py

```diff
@@ -152,28 +159,11 @@ def validate_pawn_move(
 
     def get_pawn_moves(self, pawn: models.Piece) -> typing.Iterator[int, int]:
         """Get all possible moves for a pawn."""
-        in_front = self.get_piece(pawn.rank + pawn.side.forwards, pawn.file)
-        if not in_front:
-            yield pawn.file, pawn.rank + pawn.side.forwards
-        if (not pawn.has_moved) and not in_front:
-            two_in_front = self.get_piece(
-                pawn.rank + pawn.side.forwards * 2, pawn.file
-            )
-            if two_in_front:
-                yield pawn.file, pawn.rank + pawn.side.forwards * 2
-        for direction in (-1, 1):
-            file = pawn.file + direction
-            rank = pawn.rank + pawn.side.forwards
-            target = self.get_piece(rank, file)
-            if not target:
-                en_passant_pawn = self.get_piece(pawn.rank, file)
-                en_passant_valid = (
-                    en_passant_pawn and en_passant_pawn.side != pawn.side
-                    and en_passant_pawn.first_move_last_turn
-                )
-                if en_passant_valid:
-                    yield rank, file
-            if target and target.side != pawn.side:
+        options = ((1, 0), (2, 0), (1, -1), (1, 1))
+        for absolute_rank_delta, file_delta in options:
+            rank = pawn.rank + absolute_rank_delta * pawn.side.forwards
+            file = pawn.file + file_delta
+            if validate_move_piece(self, pawn, rank, file):
```

Comment:
```suggestion
            if self.validate_move_piece(pawn, rank, file):
```
Cause that's how python works :wink: